src/stdlib_experimental_error.f90

```diff
@@ -1,34 +1,63 @@
 module stdlib_experimental_error
-use, intrinsic :: iso_fortran_env, only: stderr=>error_unit
+use, intrinsic :: iso_fortran_env, only: stderr => error_unit
+use stdlib_experimental_optval, only: optval
 implicit none
 private
 
 interface ! f{08,18}estop.f90
-module subroutine error_stop(msg, code)
-character(*), intent(in) :: msg
-integer, intent(in), optional :: code
-end subroutine error_stop
+    module subroutine error_stop(msg, code)
+        character(*), intent(in) :: msg
+        integer, intent(in), optional :: code
+    end subroutine error_stop
 end interface
 
-public :: assert, error_stop
+public :: check, error_stop
 
 contains
 
-subroutine assert(condition, code)
-! If condition == .false., it aborts the program.
-!
-! Arguments
-! ---------
-!
-logical, intent(in) :: condition
-integer, intent(in), optional :: code
-!
-! Example
-! -------
-!
-! call assert(a == 5)
-
-if (.not. condition) call error_stop("Assert failed.", code)
-end subroutine
-
-end module
+subroutine check(condition, msg, code, warn)
+
+    ! Checks the value of a logical condition. If condition == .false. and:
+    !
+    !   * No other arguments are provided, it stops the program with the default
+    !     message and exit code 1;
+    !   * msg is provided, it prints the value of msg;
+    !   * code is provided, it stops the program with the given exit code;
+    !   * warn is provided and .true., it doesn't stop the program and prints
+    !   * the message.
+    !
+    ! Arguments
+    ! ---------
+
+    logical, intent(in) :: condition
+    character(*), intent(in), optional :: msg
+    integer, intent(in), optional :: code
+    logical, intent(in), optional :: warn
+    character(*), parameter :: msg_default = 'Test failed.'
```

Comment:
Wouldn't it make more sense if the default message is `'Check failed.' just to be consistent with the subroutine name?